src/core/ext/filters/client_channel/resolver/dns/event_engine/event_engine_client_channel_resolver.cc

```diff
@@ -0,0 +1,517 @@
+  srv_handle_.reset();
+  if (!srv_records.ok()) {
+    // An error has occurred, finish resolving.
+    errors_.AddError(srv_records.status().message());
+    result = OnResolvedLocked();
+    return;
+  }
+  // Do a subsequent hostname query since SRV records were returned
+  for (const auto& srv_record : *srv_records) {
+    balancer_hostname_handles_.insert(event_engine_resolver_->LookupHostname(
+        [host = srv_record.host,
+         self = Ref(DEBUG_LOCATION, "OnBalancerHostnamesResolved")](
+            absl::StatusOr<std::vector<EventEngine::ResolvedAddress>>
+                new_balancer_addresses) {
+          self->OnBalancerHostnamesResolved(host,
+                                            std::move(new_balancer_addresses));
+        },
+        srv_record.host, std::to_string(srv_record.port),
+        resolver_->query_timeout_ms_));
+    GRPC_EVENT_ENGINE_RESOLVER_TRACE(
+        "DNSResolver::%p Started resolving balancer hostname %s:%d",
+        resolver_.get(), srv_record.host.c_str(), srv_record.port);
+  }
+}
+
+void EventEngineClientChannelDNSResolver::EventEngineDNSRequestWrapper::
+    OnBalancerHostnamesResolved(
+        std::string authority,
+        absl::StatusOr<std::vector<EventEngine::ResolvedAddress>>
+            new_balancer_addresses) {
+  ValidationErrors::ScopedField field(&errors_, "balancer hostname lookup");
+  absl::optional<Resolver::Result> result;
+  auto cleanup = absl::MakeCleanup([&]() {
+    if (result.has_value()) {
+      resolver_->OnRequestComplete(std::move(*result));
+    }
+  });
+  MutexLock lock(&on_resolved_mu_);
+  if (orphaned_) return;
+  ++number_of_balancer_hostnames_resolved_;
+  if (!new_balancer_addresses.ok()) {
+    // An error has occurred, finish resolving.
+    errors_.AddError(absl::StrCat(authority, ": ",
+                                  new_balancer_addresses.status().message()));
+  } else {
+    // Capture the addresses and finish resolving.
+    balancer_addresses_.reserve(balancer_addresses_.size() +
+                                new_balancer_addresses->size());
+    auto srv_channel_args =
+        resolver_->channel_args().Set(GRPC_ARG_DEFAULT_AUTHORITY, authority);
+    for (const auto& addr : *new_balancer_addresses) {
+      balancer_addresses_.emplace_back(CreateGRPCResolvedAddress(addr),
+                                       srv_channel_args);
+    }
+  }
+  result = OnResolvedLocked();
+}
+
+void EventEngineClientChannelDNSResolver::EventEngineDNSRequestWrapper::
+    OnTXTResolved(absl::StatusOr<std::string> service_config) {
+  ValidationErrors::ScopedField field(&errors_, "txt lookup");
+  absl::optional<Resolver::Result> result;
+  {
+    MutexLock lock(&on_resolved_mu_);
+    if (orphaned_) return;
+    GPR_ASSERT(txt_handle_.has_value());
+    txt_handle_.reset();
+    if (!service_config.ok()) {
+      errors_.AddError(service_config.status().message());
+    }
+    service_config_json_ = std::move(service_config);
```

Comment:
Similar to my comment above, [gRFC A2](https://github.com/grpc/proposal/blob/master/A2-service-configs-in-dns.md#encoding-in-dns-txt-records) says that the content of the TXT record will start with `grpc_config=`.  We're not stripping off that prefix here, which implies that we're assuming that this will be stripped off by the EE impl before returning.  However, since the EE API is supposed to be a generic mechanism for TXT record lookups, not something specific to gRPC service config, I think the EE impl should instead return that prefix as part of the result, and we should strip it off here.

CC @yijiem

Reply:
Done